src/Main.elm

```diff
@@ -41,25 +147,151 @@ update msg model =
 view : Model -> Html Msg
 view model =
     div []
-        [ div [ class "book-list" ] (List.map book model.books)
+        [ div [ class "book-list" ] (List.map (\b -> bookComponent b model.cart) model.books)
+        , div [ class "cart" ] [ cartView model ]
         ]
 
 
-book : Book -> Html Msg
-book b =
+inCart : BookID -> Cart -> Bool
+inCart bookID cart =
+    List.any (\cartItem -> cartItem.id == bookID) cart
+
+
+bookComponent : Book -> Cart -> Html Msg
+bookComponent book cart =
+    let
+        ( buttonClass, clickEvent, buttonText ) =
+            if inCart book.bookID cart then
+                ( "btn remove", Remove book.bookID, "Remove" )
+
+            else
+                ( "btn add", Add book.bookID, "Add to cart" )
+    in
     div [ class "book" ]
-        [ div [] [ img [ src <| "%PUBLIC_URL%" ++ b.cover, alt b.title ] [] ]
+        [ div [] [ img [ src <| "%PUBLIC_URL%" ++ book.cover, alt book.title ] [] ]
         , div [ class "book-info" ]
-            [ div [ class "book-title" ] [ text b.title ]
-            , div [ class "book-year" ] [ text <| String.fromInt b.year ]
-            , div [ class "book-title-CN" ] [ text b.titleCN ]
-            , div [ class "book-title-pinyin" ] [ text <| " (" ++ b.titlePinyin ++ ")" ]
-            , div [ class "book-price" ] [ text <| "$" ++ String.fromInt b.price ]
+            [ div [ class "book-title" ] [ text book.title ]
+            , div [ class "book-year" ] [ text <| String.fromInt book.year ]
+            , div [ class "book-title-CN" ] [ text book.titleCN ]
+            , div [ class "book-title-pinyin" ] [ text <| " (" ++ book.titlePinyin ++ ")" ]
+            , div [ class "book-price" ] [ text <| "$" ++ String.fromInt book.price ]
+            ]
+        , div [ class "buttons" ]
+            [ button
+                [ class buttonClass
+                , onClick clickEvent
+                ]
+                [ text buttonText
+                ]
+            , button [ onClick (Decrement book.bookID) ] [ text "-" ]
+            , text (String.fromInt (getQuantity cart book.bookID))
+            , button [ onClick (Increment book.bookID) ] [ text "+" ]
             ]
         ]
 
 
 
+-- cartView : Model -> Html Msg
+-- cartView model =
+--     div []
+--         [ div []
+--             [ if List.isEmpty model.cart then
+--                 div [ class "remove" ]
+--                     [ text "Cart is empty" ]
+--
+--               else
+--                 div [] (List.map (\c -> cartItemView c model.books) model.cart)
+--             ]
+--         , div [] [ text <| "Subtotal: $" ++ String.fromInt (cartTotalView model) ]
+--         ]
+
+
+cartView : Model -> Html Msg
+cartView model =
+    div []
+        [ table []
```

Comment:
I'm not feeling the use of a table here, I don't feel like semantically it is really a table?

I think it could even be more simple, I'll sketch up something if you like.